lib/matplotlib/animation.py

```diff
@@ -570,7 +570,9 @@ class FFMpegBase:
 
     @property
     def output_args(self):
-        args = ['-vcodec', self.codec]
+        args = []
+        if not self.outfile.endswith('.gif'):
```

Comment:
```suggestion
        if not Path(self.outfile).suffix == '.gif':
```

We take `pathlib.Path` objects as input which is what is failing the tests.

Reply:
Is this functionally different? More general because `outfile` can be something other than `str`? Committing the change, just curious for my own edification.

Reply:
`self.outfile` can be either `pathlib.Path` or `str` (which is something we test).  Either we need to agressively up-convert `outfile` to a `Path` when in comes in, or we can do a temporary conversion here.